controllers/grafanaalertrulegroup_controller.go

```diff
@@ -114,20 +114,19 @@ func (r *GrafanaAlertRuleGroupReconciler) Reconcile(ctx context.Context, req ctr
 		setNoMatchingInstance(&group.Status.Conditions, group.Generation, "ErrFetchingInstances", fmt.Sprintf("error occurred during fetching of instances: %s", err.Error()))
 		meta.RemoveStatusCondition(&group.Status.Conditions, conditionAlertGroupSynchronized)
 		r.Log.Error(err, "could not find matching instances")
-		return ctrl.Result{RequeueAfter: RequeueDelay}, err
+		return ctrl.Result{}, err
 	}
 
 	if len(instances) == 0 {
 		meta.RemoveStatusCondition(&group.Status.Conditions, conditionAlertGroupSynchronized)
 		setNoMatchingInstance(&group.Status.Conditions, group.Generation, "EmptyAPIReply", "Instances could not be fetched, reconciliation will be retried")
-		return ctrl.Result{}, nil
+		return ctrl.Result{}, fmt.Errorf("no instances found")
 	}
 
 	removeNoMatchingInstance(&group.Status.Conditions)
 	folderUID := r.GetFolderUID(ctx, group)
 	if folderUID == "" {
-		// error is already set in conditions
-		return ctrl.Result{}, nil
+		return ctrl.Result{}, fmt.Errorf("folder uid not found")
```

Comment:
I'm looking at my phone so I haven't looked at the rest of the code. But what happens if a folder is created after the alert CR?

Reply:
By returning an error, the reconciler will go into exponential backoff mode and continously retry. I'll also try to add folders as a dependency for the alertrulegroup reconciler to force a reconcile whenever a folder is created/modified

Reply:
Works even without the extra watch :partying_face: